### ISSUE TITLE
Childclass interface

### ISSUE THREAD

**Issue by calvnleong** [nus-cs2030/2223-s2#92]
# Description

Does a child class have to implement the interfaces that it's parent class implements?
Also, do I have to write 'childclass extends parentclass implements interface' or is 'childclass extends parentclass' sufficient if I want the childclass to implement the interface?
## Topic:

Specific Topics this question is related to (If Any)

## Screenshots

Screenshots (if any)

**Comment by edmundnwl** [comment]
correct me if I'm wrong but childclass extends parentclass is sufficient

when you do childclass extends parentclass implements interface, you want to implement additional interfaces that the parentclass doesn't already implement

**Comment by shakerfries** [comment]
idt a childclass have to implement the interfaces that its parent class implements

**Comment by trwstin** [comment]
why not fire up jshell and test it out? f around and find out :)

**Comment by wadeyman** [comment]
i think if you want multiple parent interfaces, you can also use childclass extends parent1, parent2 ... {

**Comment by trwstin** [comment]
> i think if you want multiple parent interfaces, you can also use childclass extends parent1, parent2 ... {

Correction: A class can only extend ONE parent class. A class can, however, extend multiple **interfaces**.

**Comment by emmannyyy** [comment]
Awesome question, really got me thinking. 

I believe there is no need for the child class to implement the interface again. Here is the logic:

1. For a parent class that implements an interface to compile, it **must** have a concrete implementation of the method. (Similar to what is taught in lectures about being a 'contract').

2. So, given that a parent class has an concrete implementation of the methods, a Child class that extends the parent class will **automatically** inherit all its methods, including those that are demanded by the interface

3. Thus, the child class does not need to implement the interface again. 

However, if you wish to explicitly know what 'contracts' the child class, I guess it doesn't hurt to force it to implement the interface again. 


TLDR
if Parent implement interface -> methods exist in parent -> child will inherit all methods -> redundant declaration for child class

**Comment by Delocy** [comment]
no implementation of the parent class is sufficient

**Comment by SimonSeeYongsheng** [comment]
Since the parent class has the methods that the interface wants to implement, the child class can then inherit the methods from the parent class.

**Comment by AmanDesai34** [comment]
idt you would need to as parent class will be able to inherit the classes to the child class. So just the parent class should be good if i Understand correctly
